migrations/20170115_api_date_validation.js

```diff
@@ -0,0 +1,73 @@
+var migrationName = '20161230_nye_hats.js';
+var authorName = 'sprusr'; // in case script author needs to know when their ...
+var authorUuid = 'db711097-c5d0-4d25-8e5e-d955da4b79fd'; //... own data is done
+
+/*
+ * API date validation
+ */
+
+var mongo = require('mongoskin');
+
+var connectionString = 'mongodb://localhost:27017/habitrpg?auto_reconnect=true'; // FOR TEST DATABASE
+
+var dbTasks = mongo.db(connectionString).collection('tasks');
+
+var query = {
+  $nor: [
+    { date: { $type: "date" } },
+    { date: { $exists: false } }
+  ]
+};
+
+console.warn('Updating tasks...');
+
+dbTasks.findEach(query, { batchSize: 250 }, function(err, task) {
+  if (err) { return exiting(1, 'ERROR! ' + err); }
+  if (!task) {
+    console.warn('All appropriate tasks found and modified.');
+    return exiting(0);
+  }
+
+  console.log(task);
```

Comment:
Ah yes, well spotted. This was indeed for testing purposes, will remove. Thanks.